eogrow/utils/validators.py

```diff
@@ -51,34 +51,44 @@ def optional_validator(value, values, config, field):  # type: ignore[no-untyped
     return validator(field, allow_reuse=allow_reuse, **kwargs)(optional_validator)
 
 
-def ensure_exactly_one_defined(param1: str, param2: str, allow_reuse: bool = True, **kwargs: Any) -> classmethod:
-    """A root validator that makes sure only one of the two parameters is defined."""
+def ensure_exactly_one_defined(first_param: str, second_param: str, **kwargs: Any) -> classmethod:
+    """A validator (applied to `second_param` field) that makes sure only one of the two parameters is defined.
 
-    def ensure_exclusion(cls: type, values: RawSchemaDict) -> RawSchemaDict:
-        is_param1_defined = values.get(param1) is None
-        is_param2_defined = values.get(param2) is None
+    Make sure that the definition of `second_param` comes after `first_param˙ (in line-order).
+    """
+
+    def ensure_exclusion(cls: type, value: Optional[Any], values: RawSchemaDict) -> Optional[Any]:
+        is_param1_defined = values.get(first_param) is None
+        is_param2_defined = value is None
         assert (
             is_param1_defined != is_param2_defined
-        ), f"Exactly one of parameters `{param1}` and `{param2}` has to be specified."
+        ), f"Exactly one of parameters `{first_param}` and `{second_param}` has to be specified."
+
+        return value
 
-        return values
+    ensure_exclusion.__name__ = f"cannot_be_used_with_{first_param}"  # used for docbuilding purposes
```

Comment:
in most cases we have a single underline used in the validators, should it be used here and in L89 as well?

Reply:
we never really decided on whether the underscore should be used. I used the underscore so that it is more clear what the difference between a validator and a field is, but all of these utility functions result in non-underscored validators in docs. From the docs perspective the underscores shouldn't be added.